### PR TITLE
RavenDB-19254 - Sharding - duplicated conflicts

### PR DESCRIPTION
**Pull request by shiranshalom**
### Issue link

https://issues.hibernatingrhinos.com/issue/RavenDB-19254/Sharding-duplicated-conflicts

### Type of change

- Bug fix

### How risky is the change?

- Low 

### Backward compatibility

- Non breaking change

### Is it platform specific issue?

- No

### Documentation update

- No documentation update is needed 

### Testing 

- It has been verified by manual testing

### Is there any existing behavior change of other features due to this change?

- No

### UI work

- No UI work is needed

### CODE REVIEW

**Comment by ppekrol** [comment]
It does not compile?

File: test/SlowTests/Sharding/Client/Operations/GetDocumentConflictsTests.cs [hunk 1]
```diff
@@ -60,13 +60,13 @@ public async Task CanGetReplicationDocumentConflicts()
                     Assert.Equal(expected, totalResults);
 
                     var documentsConflict = await store.Maintenance.SendAsync(new GetConflictsOperation(start: 0));
-                    Assert.Equal(20, documentsConflict.Results.Length);
+                    Assert.Equal(expected, documentsConflict.Results.Count);
 
-                    documentsConflict = await store.Maintenance.SendAsync(new GetConflictsOperation(start: 12));
-                    Assert.Equal(8, documentsConflict.Results.Length);
+                    documentsConflict = await store.Maintenance.SendAsync(new GetConflictsOperation(start: 2));
```
Comment (karmeli87) [inline test/SlowTests/Sharding/Client/Operations/GetDocumentConflictsTests.cs]: `pageSize` change I understand, but why the `start` reduced to `2` from `12`?

Comment (shiranshalom) [reply test/SlowTests/Sharding/Client/Operations/GetDocumentConflictsTests.cs]: because now we have 10 unique results instead of 20

**Review by ppekrol — APPROVED** [review]

File: src/Raven.Server/Documents/ConflictsStorage.cs [hunk 2]
```diff
@@ -110,30 +110,26 @@ internal GetConflictsPreviewResult GetConflictsPreviewResult(DocumentsOperationC
         {
             var table = context.Transaction.InnerTransaction.OpenTable(ConflictsSchema, ConflictsSlice);
 
-            var conflicts = new List<GetConflictsPreviewResult.ConflictPreview>();
-            foreach (var tvr in table.SeekBackwardFromLast(ConflictsSchema.FixedSizeIndexes[AllConflictedDocsEtagsSlice], skip))
+            var conflictsDictionary = new Dictionary<string, GetConflictsPreviewResult.ConflictPreview>();
+            foreach (var tvr in table.SeekBackwardFromLast(ConflictsSchema.FixedSizeIndexes[AllConflictedDocsEtagsSlice]))
```
Comment (karmeli87) [inline src/Raven.Server/Documents/ConflictsStorage.cs]: you will scan here through the entire table.
1. we need page size
2. we need skip

Skip is not the number of the unique Ids, but should be the number of scanned items.

**Review by karmeli87 — CHANGES_REQUESTED** [review]

File: src/Raven.Server/Documents/ConflictsStorage.cs (hunk 2 above)
Comment (shiranshalom) [reply src/Raven.Server/Documents/ConflictsStorage.cs]: Done

**Comment by shiranshalom** [comment]
@ml054 Changes required in the Studio:
Calculating next `Start`:
`Start` = **Sum**(`Results.ScannedResults`) + 1

File: src/Raven.Server/Documents/Sharding/Handlers/Processors/Replication/ShardedReplicationHandlerProcessorForGetConflicts.cs [hunk 3]
```diff
@@ -59,18 +58,27 @@ public ShardedGetReplicationConflictsOperation(ShardedDatabaseRequestHandler han
...
                 {
-                    final.TotalResults += s.TotalResults;
+                    final.Results.Add(res.Item);
+                    pageSize--;
+
+                    if (pageSize <= 0)
+                        break;
+
+                    var shard = res.Shard;
+                    _token.Pages[shard].Start += (int)res.Item.ScannedResults;
```
Comment (karmeli87) [inline src/Raven.Server/Documents/Sharding/Handlers/Processors/Replication/ShardedReplicationHandlerProcessorForGetConflicts.cs:81]: is `+=` correct? if you collect 2 results here next time you will skip here more than intended, no?.

File: src/Raven.Server/Documents/Handlers/Processors/Replication/AbstractReplicationHandlerProcessorForGetConflicts.cs [hunk 4]
```diff
@@ -32,29 +31,27 @@ public override async ValueTask ExecuteAsync()
...
                 {
-                    if (pageSize-- <= 0)
-                        break;
-
                     array.Add(new DynamicJsonValue
                     {
                         [nameof(GetConflictsPreviewResult.ConflictPreview.Id)] = conflict.Id,
                         [nameof(GetConflictsPreviewResult.ConflictPreview.LastModified)] = conflict.LastModified,
-                        [nameof(GetConflictsPreviewResult.ConflictPreview.ConflictsPerDocument)] = conflict.ConflictsPerDocument
+                        [nameof(GetConflictsPreviewResult.ConflictPreview.ConflictsPerDocument)] = conflict.ConflictsPerDocument,
```
Comment (karmeli87) [inline src/Raven.Server/Documents/Handlers/Processors/Replication/AbstractReplicationHandlerProcessorForGetConflicts.cs]: can we have `ToJson` on the `conflict` ?

**Comment by ml054** [comment]
@shiranshalom I can see we added continuation token some time ago. So why I need to provide some special params as well? Or we removed continuation token?

**Comment by shiranshalom** [comment]
> @shiranshalom I can see we added continuation token some time ago. So why I need to provide some special params as well? Or we removed continuation token?

@ml054 We still have `ContinuationToken` :-). 
We need `Results.ScannedResults` for regular case (to calculate `Start`).

File: src/Raven.Server/Documents/Handlers/Processors/Replication/AbstractReplicationHandlerProcessorForGetConflicts.cs (hunk 4 above)
Comment (shiranshalom) [reply src/Raven.Server/Documents/Handlers/Processors/Replication/AbstractReplicationHandlerProcessorForGetConflicts.cs]: done

File: src/Raven.Server/Documents/Sharding/Handlers/Processors/Replication/ShardedReplicationHandlerProcessorForGetConflicts.cs (hunk 3 above)
Comment (shiranshalom) [reply src/Raven.Server/Documents/Sharding/Handlers/Processors/Replication/ShardedReplicationHandlerProcessorForGetConflicts.cs:81]: You are right. Done.

**Review by karmeli87 — APPROVED** [review]

**Comment by ml054** [comment]
@shiranshalom let's move scanned results to top level (next to total results field) to mimic behavior we have in query results.

**Comment by ml054** [comment]
@shiranshalom I did my changes in blind - so either please test studio paging for sharded and non-sharded dbs or let me know once you finish your changes and I will test.

**Comment by shiranshalom** [comment]
> @shiranshalom let's move scanned results to top level (next to total results field) to mimic behavior we have in query results.

@ml054 Done

**Comment by shiranshalom** [comment]
> @shiranshalom I did my changes in blind - so either please test studio paging for sharded and non-sharded dbs or let me know once you finish your changes and I will test.

Tested manually for both Sharded and Non-sharded.

**Review by ppekrol — APPROVED** [review]

**Pull request merged**